src/transactionsFactories/smartContractTransactionsFactory.ts

```diff
@@ -80,25 +84,53 @@ export class SmartContractTransactionsFactory {
 
     createTransactionForExecute(options: {
         sender: IAddress,
-        contractAddress: IAddress,
+        contract: IAddress,
         functionName: string,
         gasLimit: BigNumber.Value,
-        args?: any[]
-    }
-    ): DraftTransaction {
+        args?: any[],
+        nativeTransferAmount?: BigNumber.Value,
+        tokenTransfers?: TokenTransfer[]
+    }): DraftTransaction {
         const args = options.args || [];
-        let parts: string[] = [options.functionName];
+        const tokenTransfer = options.tokenTransfers || [];
+        const nativeTransferAmount = options.nativeTransferAmount ?? 0;
+        const numberOfTokens = tokenTransfer.length;
 
-        const preparedArgs = this.argsToDataParts(args, this.abiRegistry?.getEndpoint(options.functionName));
-        parts = parts.concat(preparedArgs);
+        if (nativeTransferAmount && numberOfTokens) {
+            throw new ErrBadUsage("Can't send both native token and ESDT/NFT tokens");
+        }
+
+        let receiver = options.contract;
+        const tokenComputer = new TokenComputer();
```

Comment:
Maybe it will be received as a constructor parameter (see comment for sdk-py).

Reply:
It's now received as an interface, same as in sdk-py-core.